egs/wsj/s5/steps/nnet3/tdnn/train.sh

```diff
@@ -5,7 +5,7 @@
 # Copyright 2012-2015  Johns Hopkins University (Author: Daniel Povey).
 #           2013  Xiaohui Zhang
 #           2013  Guoguo Chen
-#           2014  Vimal Manohar
+#           2014-2016  Vimal Manohar
```

Comment:
Vimal, I don't want to check in this code as-is.
You have made a huge number of changes to nnet3/tdnn/train.sh.  This creates problems for testing.
Obviously we're not going to re-run the million TDNN training setups that we have here and there in Kaldi.
And even apart from that, your changes obscure what was already a very complicated script.
I would possibly be OK with checking in a separate script whose purpose is to train DNNs for this type of setup.
Or you may be able to do it with Vijay's train_dnn.py, although again, I don't want a situation where you introduce
a ton of if statements into a program-- if there are a lot of changes, I'd prefer you to create a separate python version of the training script; this will mean we don't have to worry about introducing bugs into Vijay's working setup.
I don't have strong opinions about whether you check in a shell script or a python script, but I think it should be a separate script.  Right now your changes add to much complexity to the script.